packages/pymedphys_deliverydata/tests/deliverydata/test_dicom_round_trip.py

```diff
@@ -26,34 +26,81 @@
 
 import os
 
+import pytest
+
 import numpy as np
 import matplotlib.pyplot as plt
 
 import pydicom
 
-from pymedphys_core.deliverydata.dicom import (
-    dicom_to_delivery_data, delivery_data_to_dicom,
-    get_gantry_angles_from_dicom, maintain_order_unique,
-    filter_out_irrelevant_control_points)
+from pymedphys_utilities.algorithms import maintain_order_unique
 
 from pymedphys_fileformats.trf import delivery_data_from_logfile
+from pymedphys_dicom.rtplan import (
+    get_metersets_from_dicom,
+    get_gantry_angles_from_dicom)
+
+from pymedphys_mudensity.mudensity import mu_density_from_delivery_data
+
+from pymedphys.deliverydata import DeliveryData
+
+# pylint: disable=redefined-outer-name
+
 
-from pymedphys_core.mudensity import mu_density_from_delivery_data
+DIR_TO_TEST_MAP = {
+    "original": {
+        'fraction_group': 1
+    },
+    "multi_fraction_groups": {
+        'fraction_group': 2
+    }
+}
 
+DIR_TO_TEST = 'original'
+FRACTION_GROUP = DIR_TO_TEST_MAP[DIR_TO_TEST]['fraction_group']
 
 DATA_DIRECTORY = os.path.join(
-    os.path.dirname(__file__), "data")
+    os.path.dirname(__file__), "data", DIR_TO_TEST)
 DICOM_FILEPATH = os.path.abspath(os.path.join(
-    DATA_DIRECTORY, "RP.2.16.840.1.114337.1.1.1548043901.0_Anonymised.dcm"))
+    DATA_DIRECTORY, "rtplan.dcm"))
 LOGFILE_FILEPATH = os.path.abspath(os.path.join(
     DATA_DIRECTORY, "imrt.trf"))
 
 
-def test_mudensity_agreement():
-    dicom = pydicom.dcmread(DICOM_FILEPATH, force=True)
-    dicom_delivery_data = dicom_to_delivery_data(dicom)
+@pytest.fixture
+def loaded_dicom_dataset():
+    return pydicom.dcmread(DICOM_FILEPATH, force=True)
+
+
+@pytest.fixture
+def logfile_delivery_data() -> DeliveryData:
+    return delivery_data_from_logfile(LOGFILE_FILEPATH)
```

Comment:
Just putting it out there, fixtures are awesome! ... I am hooked...

Reply:
Agree!